controllers/glanceapi_controller.go

```diff
@@ -1545,11 +1547,13 @@ func (r *GlanceAPIReconciler) ensureGlanceAPITopology(
 			tpRef.Namespace = helper.GetBeforeObject().GetNamespace()
 		}
 		// Retrieve the referenced Topology
+		defaultLabelSelector := glanceapi.GetGlanceAPILabelSelector(instance)
```

Comment:
This should directly call `labels.GetSingleLabelSelector(` and pass down a struct to this call. This is important because if you look at other PRs, where we have multiple Components (manilaAPI, manilaScheduler, manilaShare) we run this same code multiple times for different `instances`.

Reply:
Done